### PR TITLE
tfgen for public images

### PR DESCRIPTION
**Pull request by imjasonh**
This moves `tfgen` into a subcommand of `monopod`, and adds `make tfgen` and related targets to the Makefile.

Because `busybox`, `git` and `static` are weird and have Alpine-based images, we handle these separately in this repo, and just skip them entirely, and hand-write the output for them. This isn't great, but it's progress.

This should hopefully let us remove some code from `monopod scaffold`, and make broad changes easier to script.

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]

File: images/terraform/generated.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "ref"    = module.busl.image_ref
+        "config" = module.busl.config
+        "tags"   = ["latest"]
+      }
+    },
+    {
+      basename(path.module) = {
+        "ref"    = module.mpl.image_ref
+        "config" = module.mpl.config
+        "tags"   = ["each.key"]
```
Comment (jdolitsky) [inline images/terraform/generated.tf:16]: ...

Comment (jdolitsky) [reply images/terraform/generated.tf]: added a check for this

File: images/metallb/main.tf [hunk 2]
```diff
@@ -47,7 +47,7 @@ resource "oci_tag" "latest" {
 resource "oci_tag" "latest-dev" {
   for_each   = local.components
   depends_on = [module.test-latest]
-  digest_ref = module.latest[each.key].image_ref
+  digest_ref = module.latest[each.key].dev_ref
```
Comment (jdolitsky) [inline images/metallb/main.tf:50]: unrelated, but found during this

**Review by jdolitsky — APPROVED** [review]

**Review by jdolitsky — APPROVED** [review]

**Review by luhring — APPROVED** [review]

**Comment by imjasonh** [comment]
Force merging past flaky tests

**Pull request merged**